.circleci/config.yml

```diff
@@ -1,150 +1,91 @@
-            tag=${app}.${version}
-
-            echo export CIRCLE_BRANCH_LOWERCASE=${CIRCLE_BRANCH_LOWERCASE} > /tmp/mtp-env.sh
-            echo export registry=${registry} >> /tmp/mtp-env.sh
-            echo export version=${version} >> /tmp/mtp-env.sh
-            echo export tag=${tag} >> /tmp/mtp-env.sh
-
-            echo "Building ${tag}"
-      - setup_remote_docker:
-          docker_layer_caching: true
-      - run:
-          name: Log into ECR
-          command: |
-            source /tmp/mtp-env.sh
-            aws ecr get-login-password --region ${AWS_DEFAULT_REGION} | docker login --username AWS --password-stdin ${ECR_ENDPOINT}
-      - run:
-          name: Pull base image
-          command: |
-            source /tmp/mtp-env.sh
-            docker pull ${registry}:base-web
-            docker tag ${registry}:base-web base-web
-      - run:
-          name: Build docker image
-          command: |
-            source /tmp/mtp-env.sh
-            docker build \
-              --force-rm \
-              --build-arg APP_GIT_COMMIT=${CIRCLE_SHA1} \
-              --build-arg APP_GIT_BRANCH=${CIRCLE_BRANCH} \
-              --build-arg APP_BUILD_TAG=${tag} \
-              --build-arg APP_BUILD_DATE=$(date +%FT%T%z) \
-              --tag ${tag} \
-              .
-      - run:
-          name: Test docker image
-          command: |
-            source /tmp/mtp-env.sh
-            docker run \
-              --name postgres \
-              --detach \
-              -e POSTGRES_USER=postgres \
-              -e POSTGRES_PASSWORD=postgres \
-              -e POSTGRES_DB=mtp_api \
-              circleci/postgres:10.10-alpine
-            sleep 10
-            docker run \
-              --name ${app} \
-              -e DJANGO_SETTINGS_MODULE=mtp_api.settings.ci \
-              -e DB_PASSWORD=postgres \
-              -e DB_USERNAME=postgres \
-              -e DB_HOST=postgres \
-              --link postgres \
-              ${tag} \
-              /bin/bash -c '/app/venv/bin/pip install -r requirements/ci.txt && /app/venv/bin/python manage.py test --verbosity=2'
-      - run:
-          name: Extract test reports
-          when: always
-          command: |
-            source /tmp/mtp-env.sh
+    - checkout
+    - setup_remote_docker:
+        docker_layer_caching: true
+    - run:
+        name: Build docker image
+        command: . ${CIRCLE_WORKING_DIRECTORY}/.circleci/custom_scripts/build_service_image.sh
+    - persist_to_workspace:
+        root: imagedump
+        paths:
+          - '*.tar.gz'
+  test:
+    executor: build-test-push
+    parallelism: 16
+    steps:
+    - checkout
+    - setup_remote_docker:
+        docker_layer_caching: true
+    - attach_workspace:
+        at: imagedump
+    - run:
+        name: Test docker image
+        command: . ${CIRCLE_WORKING_DIRECTORY}/.circleci/custom_scripts/test.sh
+    - run:
+        name: Extract test reports
+        when: always
+        command: |
```

Comment:
maybe this should also be in a separate file so that _all_ shell scripts are files

Reply:
I didn't think it was worth it in this case, as it's just 4 lines, none of which rely on dynamically templated environment variables. But I can move it out if you want